homeassistant/components/mqtt/discovery.py

```diff
@@ -98,8 +106,20 @@
 
             _LOGGER.info("Found new component: %s %s", component, discovery_id)
 
-            await async_load_platform(
-                hass, component, platform, payload, hass_config)
+            if platform in CONFIG_ENTRY_PLATFORMS.get(component, []):
```

Comment:
Can you reverse this `if`, call `async_load_platform` and then return. Saves on indentation.